capabilities/container/container.go

```diff
@@ -0,0 +1,113 @@
+package container
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/docker/go-connections/nat"
+	"github.com/puppetlabs/lumogon/capabilities/payloadfilter"
+	"github.com/puppetlabs/lumogon/capabilities/registry"
+	"github.com/puppetlabs/lumogon/dockeradapter"
+	"github.com/puppetlabs/lumogon/logging"
+	"github.com/puppetlabs/lumogon/types"
+)
+
+var containerDescription = `The 'container' capability captures detailed container information`
+
+var containerCapability = dockeradapter.DockerAPICapability{
+	Capability: types.Capability{
+		Schema:      "http://puppet.com/lumogon/capability/container/draft-01/schema#1",
+		Title:       "Container Information",
+		Name:        "container",
+		Description: containerDescription,
+		Type:        "dockerapi",
+		Payload:     nil,
+		SupportedOS: map[string]int{"all": 1},
+	},
+	Harvest: func(capability *dockeradapter.DockerAPICapability, client dockeradapter.Harvester, id string, target types.TargetContainer) {
+		logging.Debug("[Container Info] Harvesting container information associated with %s [%s]", target.Name, target.ID)
+		capability.HarvestID = id
+
+		version, err := InspectContainer(client, target.ID)
+		if err != nil {
+			capability.PayloadError(err.Error())
+			return
+		}
+
+		filtered, _ := payloadfilter.Filter(version)
+
+		capability.Payload = filtered
+	},
+}
+
+// InspectContainer Extracts and returns a formatted map[string]interface{} containing
+// a subset of information returned by ContainerInspect
+func InspectContainer(client dockeradapter.Harvester, targetID string) (map[string]interface{}, error) {
+	ctx := context.Background()
+	c, err := client.ContainerInspect(ctx, targetID)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO - **IMPORTANT** this contains only a subset of the information available
+	// it explicitly avoids including any structured data (ports/mappings etc) pending
+	// support in the UI, it also avoids any config that could potentially contain
+	// sensitive data.
+
+	result := map[string]interface{}{
+		"Hostname":           c.Config.Hostname,
+		"Domainname":         c.Config.Domainname,
+		"User":               c.Config.User,
+		"Image":              c.Config.Image,
+		"AttachStdin":        fmt.Sprintf("%t", c.Config.AttachStdin),
```

Comment:
What's the rationale for converting booleans to strings? Feels like these would be useful as native JSON types?

Reply:
I'm returning the int64s natively now, the bools are being returned as strings for the moment as it requires a reporter UI change, will address in a separate PR.